src/test/java/GmailPage.java

```diff
@@ -0,0 +1,243 @@
+    @FindBy(css = "div.G-tF div.Bn")
+    private WebElement buttonDeleteEmailForever;
+
+    public void waitUntilLinkToLoggedUserPresent() {
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector(LINK_TO_LOGGED_USER)));
+    }
+
+    public boolean loggedUserContainsLoginUsername(String loginUsername) {
+        return linkToLoggedUser.getAttribute(ATTRIBUTE_LOGGED_USER).contains(loginUsername);
+    }
+
+    public void waitUntilMaskPreventingClickDisappear() {
+        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.cssSelector(MASK_PREVENTING_CLICK)));
+    }
+
+    public void waitUntilButtonToDelSelectedAcctPresent() {
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector(BUTTON_TO_DELETE_SELECTED_ACCOUNT)));
+    }
+
+    public void waitUntilEmailToSelectAndDeletePresent() {
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector(EMAIL_TO_SELECT_AND_DELETE)));
+    }
+
+    public void waitUntilButtonToConfirmEmailDeletionPresent() {
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector(BUTTON_TO_CONFIRM_EMAIL_DELETION)));
+    }
+
+    public LoginPage logout() {
+        linkToLoggedUser.click();
+        linkToSignOut.click();
+        LoginPage loginPage = new LoginPage(driver, wait);
+        dropDownWithEmailToExpand.click();
+        waitUntilMaskPreventingClickDisappear();
+        waitUntilButtonToDelSelectedAcctPresent();
+        buttonToDeleteSelectedAccount.click();
+        waitUntilEmailToSelectAndDeletePresent();
+        emailToSelectAndDelete.click();
+        waitUntilButtonToConfirmEmailDeletionPresent();
+        buttonToConfirmEmailDeletion.click();
+        return loginPage;
+    }
+
+    public void sendEmail(String recepientEmail, String emailSubject, String emailMsgBody) {
+        linkCreateNewEmail.click();
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector(INPUT_EMAIL_TO)));
+        inputEmailTo.sendKeys(recepientEmail + MAIL_BOX_DOMAIN);
+        inputEmailSubject.sendKeys(emailSubject);
+        inputEmailMessageBody.sendKeys(emailMsgBody);
+        buttonSendEmail.click();
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector(TOOLTIP_ACTION_SUCCESSFUL)));
+        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.cssSelector(TOOLTIP_ACTION_SUCCESSFUL)));
+    }
+
+    public boolean checkEmailReceived(String emailSender, String emailSubject, String emailMsgBody) {
+        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.cssSelector(HINT_NEW_EMAIL_ARRIVED)));
+        boolean isInboxEmailCorrect = false;
+        isInboxEmailCorrect = allRowsWithInboxEmails.stream()
+                .anyMatch(singleRowWithInboxEmail -> {
+                    return ((emailSender + MAIL_BOX_DOMAIN).equals(singleRowWithInboxEmail.findElement(INBOX_EMAIL_RECEPIENT).getAttribute(ATTRIBUTE_EMAIL_RECEPIENT_TO_COMPARE_WITH))
```

Comment:
It is better to split big expressions:
boolean isSenderCorrect = ...
boolean isSubjectCorrect = ...
boolean isBodyCorrect = ...
return isSenderCorrect && isSubjectCorrect && isBodyCorrect;